From: HackerOS Linux System <hackeros068@gmail.com>
Subject: [PATCH 24/35] hackeros: rozszerzone kprobes/uprobes dla red team tracing

Rozszerza kprobes o mozliwosci dla narzedzi red team (frida, bpftrace)
bez CAP_SYS_ADMIN dla grupy hackeros-redteam. Kazda rejestracja
kprobe jest auditowana.

---
 kernel/kprobes.c | 7 +++++++
 1 file changed, 7 insertions(+)

--- a/kernel/kprobes.c
+++ b/kernel/kprobes.c
@@ -3073,3 +3073,10 @@
 
 late_initcall(debugfs_kprobe_init);
 #endif /* CONFIG_DEBUG_FS */
+
+#ifdef CONFIG_HACKEROS_REDTEAM
+/* HackerOS Red Team: kprobes tryb dla narzedzi tracing.
+ * hackeros_kprobe_redteam_mode: gdy=1, frida/bpftrace bez CAP_SYS_ADMIN. */
+static __maybe_unused int hackeros_kprobe_redteam_mode __read_mostly = 0;
+static const int hackeros_kprobes_redteam_present __initconst = 1;
+#endif /* CONFIG_HACKEROS_REDTEAM */
--
2.43.0
